core/iwasm/fast-jit/fe/jit_emit_control.c

```diff
@@ -227,11 +264,26 @@ push_jit_block_to_stack_and_pass_params(JitCompContext *cc, JitBlock *block,
         }
         else {
             /* IF block with condition br insn */
-            if (!GEN_INSN(CMP, cc->cmp_reg, cond, NEW_CONST(I32, 0))
-                || !(insn = GEN_INSN(BNE, cc->cmp_reg,
-                                     jit_basic_block_label(basic_block), 0))) {
-                jit_set_last_error(cc, "generate cond br failed");
-                goto fail;
+            if (insn_select && insn_cmp) {
```

Comment:
I thought `get_last_cmp_and_selectcc` will do the optimization. So, what does this `if(){}` do?

Reply:
get_last_cmp_and_selectcc just returns insn_select and insn_cmp found, here we replace insn_select with Bcc.